### ISSUE TITLE
Azure - Container Host Can't Remove Invalid Policies

### ISSUE THREAD

**Issue by tomarv2** [cloud-custodian/cloud-custodian#5735]
I am running policies in ACI:

**Policy:**

I want to tag resource created with the time that it was created, this is the policy I came up with:

```
  - name: auto-owner-storage-account
    resource: azure.storage
    mode:
      type: azure-event-grid
      events: [{
          resourceProvider: 'Microsoft.Storage/storageAccounts',
          event: 'write'
        }]
    filters:
      - "tag:Owner": null
    actions:
      - type: tag
        key: "DateCreated"
        value: "{now}"

```

**Error:**

```
Job "Host.update_policies (trigger: interval[0:01:00], next run at: 2020-05-08 17:13:23 UTC)" raised an exception
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/apscheduler/executors/base.py", line 125, in run_job
    retval = job.func(*job.args, **job.kwargs)
  File "/usr/local/lib/python3.7/site-packages/c7n_azure/container_host/host.py", line 166, in update_policies
    self.unload_policy_file(policy_path, policies_copy)
  File "/usr/local/lib/python3.7/site-packages/c7n_azure/container_host/host.py", line 263, in unload_policy_file
    removed = [policies.pop(p['name']) for p in policy_config.get('policies', [])]
  File "/usr/local/lib/python3.7/site-packages/c7n_azure/container_host/host.py", line 263, in <listcomp>
    removed = [policies.pop(p['name']) for p in policy_config.get('policies', [])]
KeyError: 'auto-owner-storage-account'
Pulled 5 events to process while polling queue.
Job "Host.poll_queue (trigger: interval[0:00:15], next run at: 2020-05-08 17:12:38 UTC)" executed successfully
Running job "Host.poll_queue (trigger: interval[0:00:15], next run at: 2020-05-08 17:12:53 UTC)" (scheduled at 2020-05-08 17:12:38.048172+00:00)
```

Am I missing something @logachev @stefangordon

**Comment by kapilt** [comment]
the error here seems unrelated to the policy, it looks like some sort of logic bug in the container host, as its trying to unload policies loaded in memory.

**Comment by stefangordon** [comment]
For the action, you probably want this: 
https://cloudcustodian.io/docs/azure/resources/azure-common-actions.html#auto-tag-date

Regarding the error - I actually saw this a few days ago.  I believe we may have a bug that prevents a policy from being updated if the existing version is invalid.  Basically we try to parse the old cached ones as part of the update process (to see the policy names) and get into a state that can't be resolved without restarting the container.

I'll investigate.  I'm also going to rename this issue as I think your actual error is this container host bug.

**Comment by stefangordon** [comment]
Oh, apparently I had already fixed this locally and I forgot.  I pushed the changes I had, they are probably sufficient for now without doing too much refactoring (and re-testing).

**Comment by tomarv2** [comment]
this worked, closing: 

```
auto-tag-date
```

**Comment by kapilt** [comment]
reopening, as there's a separate issue here that needs addressing with container host mode, independent of the auto-tag policy.

**Comment by stefangordon** [comment]
I believe both issues here are fixed now, the unloading failure was fixed in #5737
